### PR TITLE
Add companies cubit

### PR DESCRIPTION
**Pull request by manhduc-nguyen**

### CODE REVIEW

File: lib/blocs/companies_cubit/companies_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import 'dart:async';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:twake/blocs/companies_bloc/companies_bloc.dart';
+import 'package:twake/blocs/notification_bloc/notification_bloc.dart';
```
Comment (bobs4462) [inline lib/blocs/companies_cubit/companies_cubit.dart:5]: Will be replaced by SynchronizationCubit

File: lib/blocs/companies_cubit/companies_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import 'dart:async';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:twake/blocs/companies_bloc/companies_bloc.dart';
```
Comment (bobs4462) [inline lib/blocs/companies_cubit/companies_cubit.dart:4]: Why is this here?

File: lib/blocs/companies_cubit/companies_cubit.dart [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import 'package:twake/blocs/notification_bloc/notification_bloc.dart';
+import 'package:twake/blocs/profile_bloc/profile_bloc.dart';
```
Comment (bobs4462) [inline lib/blocs/companies_cubit/companies_cubit.dart:6]: Globals.instance now store globally available data

File: lib/blocs/companies_cubit/companies_cubit.dart [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import 'package:twake/models/globals/globals.dart';
+import 'package:twake/repositories/collection_repository.dart';
```
Comment (bobs4462) [inline lib/blocs/companies_cubit/companies_cubit.dart:8]: No need to reuse collections_repository, just create a specialized repository for this cubit, analogous to WorkspacesRepository

File: lib/blocs/companies_cubit/companies_cubit.dart [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  Future<void> clear() async {
+    await repository.clean();
+    emit(CompaniesEmpty());
+  }
+
+  void selectCompany(String companyId) {
+    Globals.instance.companyId = companyId;
+    repository.select(companyId);
+    ProfileBloc.selectedCompanyId = companyId;
+    final newState = CompaniesLoaded(
```
Comment (bobs4462) [inline lib/blocs/companies_cubit/companies_cubit.dart:54]: Also you can review the list of available states for CompaniesCubit, and remove unused ones

File: lib/repositories/workspaces_repository.dart [hunk 6]
```diff
@@ -6,6 +8,10 @@ import 'package:twake/services/service_bundle.dart';
 class WorkspacesRepository {
   final _api = ApiService.instance;
   final _storage = StorageService.instance;
+  final _workSpaceStreamController = StreamController<List<Workspace>>();
+
+  // register to this stream to get workspace updates;
+  Stream<List<Workspace>> get workSpaceStream => _workSpaceStreamController.stream;
```
Comment (bobs4462) [inline lib/repositories/workspaces_repository.dart:14]: Let's keep workspace as a single word, without camelCasing it like workSpace

**Review by bobs4462 — CHANGES_REQUESTED** [review]
Check out the comments that I left throughout the code

**Comment by manhduc-nguyen** [comment]
> Check out the comments that I left throughout the code

@bobs4462 pls check again, I think `CompaniesCubit` is just that simple for now.

File: lib/repositories/company_repository.dart [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import 'package:twake/models/company/company.dart';
+
+class CompanyRepository {
+  Future<List<Company>> fetch() async {
```
Comment (bobs4462) [inline lib/repositories/company_repository.dart:4]: Let's name methods more uniformly. For example in workspacesRepository similar method is named getWorkspaces, but fetch seems to be a little less descriptive

File: lib/blocs/companies_cubit/company_state.dart [hunk 8]
```diff
@@ -1,19 +1,25 @@
 import 'package:equatable/equatable.dart';
-import 'package:twake/models/company.dart';
+import 'package:twake/models/company/company.dart';
 
 abstract class CompaniesState extends Equatable {
   const CompaniesState();
 }
 
+class InitState extends CompaniesState {
```
Comment (bobs4462) [inline lib/blocs/companies_cubit/company_state.dart:8]: @RockinPaul can you please provide us with some uniform naming convention for states?
You can use companies_state as an example.

**Review by bobs4462 — COMMENTED** [review]
Almost done, need to consult @RockinPaul about namings

File: lib/blocs/companies_cubit/company_state.dart (hunk 8 above)
Comment (manhduc-nguyen) [reply lib/blocs/companies_cubit/company_state.dart:8]: how about `Companies_[state-name]_State` ? 
so in this case `CompaniesInitialState`

File: lib/repositories/company_repository.dart (hunk 7 above)
Comment (RockinPaul) [reply lib/repositories/company_repository.dart:4]: @bobs4462 I can't agree with you regarding "fetch" - it's a quite traditional word for the naming functions that perform requests to remote or local data sources (just try to google how popular it is actually: https://www.w3schools.com/js/js_api_fetch.asp).
For me 'get' is less descriptive and expresses simply the getter of a small variable but never asynchronous function.
I'm firmly convinced that we should stick to one style, despite a matter of habits.

But: 

изображение

Comment (RockinPaul) [reply lib/repositories/company_repository.dart:4]: @bobs4462 it's here: https://dart.dev/guides/language/effective-dart/design

File: lib/blocs/companies_cubit/company_state.dart (hunk 8 above)
Comment (RockinPaul) [reply lib/blocs/companies_cubit/company_state.dart:8]: @bobs4462 @manhduc-nguyen Felix Angelov can: https://bloclibrary.dev/#/blocnamingconventions
I often repeat this again and again keeping this page open during my work with Bloc.

File: lib/repositories/company_repository.dart (hunk 7 above)
Comment (bobs4462) [reply lib/repositories/company_repository.dart:4]: Ok, then let's use fetch for requesting data, @manhduc-nguyen please update the method name in WorkspacesRepository

File: lib/blocs/companies_cubit/company_state.dart (hunk 8 above)
Comment (bobs4462) [reply lib/blocs/companies_cubit/company_state.dart:8]: Alright let's use the conventions from BLoC library creator. Please add the resource to Notion for further reference.

Comment (manhduc-nguyen) [reply lib/blocs/companies_cubit/company_state.dart]: Nice, fixed.

**Review by bobs4462 — APPROVED** [review]
